src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -1422,6 +1421,12 @@ public static bool IsInstanceContext(this SyntaxTree syntaxTree, int position, S
                 return false;
             }
 
+            var expressionBody = GetExpressionBodyFromMember(containingMember);
+            if (expressionBody != null)
+            {
+                return ArrowExpressionClauseExpressionAdjustedSpan(expressionBody).IntersectsWith(position);
```

Comment:
I understand now, I misread the implementation of `ArrowExpressionClauseExpressionAdjustedSpan`.  With a different name I think this is fine.